### PR TITLE
GP19-159: Redirect Conditional Re-Render Bug Fix

### PR DESCRIPTION
**Pull request by MerrettM**
fixing a bug that caused the setting of a new cookie not to trigger a re-render of the `Component`/`Redirect` conditionals in `App`

### CODE REVIEW

File: src/Conference/AddConferenceRedirector.jsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+  return (
+    <>
+      {cookies.sessionToken ? (
+        <AddConference />
+      ) : (
+        <Redirect to="/users/login" />
+      )}
+    </>
+  );
```
Comment (longstaff-elsevier) [inline src/Conference/AddConferenceRedirector.jsx]: should be able to be simplified to 

`return cookies.sessionToken ? <AddConference /> : <Redirect to="/users/login" />`

?

File: src/User/UserContainerRedirector.jsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  return (
+    <>
+      {cookies.sessionToken ? (
+        <UserContainer />
+      ) : (
+        <Redirect to="/users/login" />
+      )}
+    </>
+  );
+};
```
Comment (longstaff-elsevier) [inline src/User/UserContainerRedirector.jsx]: make generic?

`
const UserRedirector = ({redirectPath, children}) => {
  const [cookies] = useCookies([cookieName]);

  return (
      cookies.sessionToken ? children : <Redirect to={redirectPath} />
`

File: src/App.js [hunk 3]
```diff
@@ -21,8 +20,6 @@ const StyledApp = styled.div`
   min-height: 100vh;
 `;
 function App() {
-  const [cookies] = useCookies([cookieName]);
```
Comment (longstaff-elsevier) [inline src/App.js]: Original bug is probably that you are referencing this outside of the provider (first line of render) so wont hook into updates

File: src/App.js [hunk 4]
```diff
@@ -39,7 +42,9 @@ function App() {
               <LoginForm />
             </Route>
             <Route path="/users/:id">
-              <UserContainerRedirector />
+              <CookieOrRedirect redirectPath="/users/login">
```
Comment (tonymaynard) [inline src/App.js]: May I ask why you chose to prefix with Cookie again? A user could have multiple cookies on their browser

Comment (tonymaynard) [reply src/App.js]: Perhaps AuthRedirect ?

Comment (MerrettM) [reply src/App.js]: Thats true, I used that one mainly because it was descriptive but AuthRedirect would probably be more accurate to it's function

**Pull request merged**